felix/iptables/table.go

```diff
@@ -780,6 +786,17 @@ func (t *Table) getHashesAndRulesFromDataplane() (hashes map[string][]string, ru
 // attemptToGetHashesAndRulesFromDataplane starts an iptables-save subprocess and feeds its output to
 // readHashesAndRulesFrom() via a pipe.  It handles the various error cases.
 func (t *Table) attemptToGetHashesAndRulesFromDataplane() (hashes map[string][]string, rules map[string][]string, err error) {
+	startTime := t.timeNow()
+	defer func() {
+		saveDuration := t.timeNow().Sub(startTime)
+		t.peakIptablesSaveTime = t.peakIptablesSaveTime * 99 / 100
```

Comment:
Is there any lower-bound to this value, or will it (slowly) tend towards lower and lower values forever?

Reply:
It'll get lower and lower, until we see a higher value.  The idea is that, if the time is the same every time, the peak will reflect that, but if it's spiky, then the peak will capture the spike and then decay slowly (but it will eventually decay if there are no more spikes).